### ISSUE TITLE
Unable to run on local machine

### ISSUE THREAD

**Issue by aloksguha** [JoshuaR503/Stock-Market-App#1]
Hi Amigos,

I am unable to run it on my local machine, could you please provide the necessary steps for the same. It looks like an exciting app to me.

Thanks,
Alok

**Comment by JoshuaR503** [comment]
Does it show any error messages? If so, could you show it to me?

**Comment by qweink** [comment]
Hi,package:sma/key.dart doesn't exist ,how to solve this problem?

**Comment by JoshuaR503** [comment]
The file key.dart contains the API keys from the services that I use. 

You need to get your own API keys from [Alpha Vantage](https://www.alphavantage.co/support/#api-key), [NewsAPI.org](https://newsapi.org/account), and [Finnhub ](https://finnhub.io/). 

For Sentry (sentry_helper.dart), you will need to create your own project and [get your own DNS](https://forum.sentry.io/t/where-can-i-find-my-dsn/4877).

**Comment by JoshuaR503** [comment]
Please pull the new changes and then follow the steps that I've included [here](https://github.com/JoshuaR503/Stock-Market-App/blob/master/lib/keys/api_keys.dart). 

I hope this helps you solve the issues that you're having. If it doesn't help, please let me know.

**Comment by fogelfish** [comment]
Hello @JoshuaR503, I had the same question.  Maybe the instructions could be modified.  To solve the missing `package:sma/key.dart` I created the `key.dart` file and in it defined the variables you require in `api_keys.dart`.  At least that resolved an undefined name **sentryDNS** in `lib/helpers/sentry_helper.dart`.

After doing that I tried to compile to iOS but encountered some errors.  Do you want me to open a new ticket for those?

**Comment by JoshuaR503** [comment]
@fogelfish, If you go the file [api_keys.dart](https://github.com/JoshuaR503/Stock-Market-App/blob/master/lib/keys/api_keys.dart), you will see that in the 3rd line there is an import of `package:sma/key.dart`. You must remove line 3 in the file api_keys.dart, which is `import 'package:sma/key.dart';`.

[Here is how to get your Sentry DNS](https://docs.sentry.io/clients/java/config/#setting-the-dsn). 

In a nutshell, you must create a project. When they ask you to choose a platform you must choose "other" because if you type "Flutter" or "Dart" you will get this message: "We don't have an SDK for that yet!". After you create a project, you will need to configure your project. This is where you can DNS. You will see this message: "Already have things setup? Get your DSN." You click it and a string will appear. You need to replace  `sentryDNS` with the string that you got from Sentry. 

I hope this helps. Let me know if you still need help! :)

**Comment by fogelfish** [comment]
@JoshuaR503, before I encountered the compilation errors (which I didn't describe) I first had to resolve the problem with the missing `key.dart`.  Following your suggestion I removed the import statement from `api_keys.dart` ...and by the way I also had to remove it from `sentry_helper.dart`.

I thought I was supposed to replace sentryDNS and the other three variables in `api_keys.dart` with the actual values I'd acquired, but after doing that I got stuck with a different error: `undefined name sentryDNS in sentry_helper.dart`.

So that's when I decided to revert to the original `api_keys.dart` and I put the import statement back where they'd been and I created my own `key.dart` file where I defined the `sentryDNS`, `alphavantageKey`, `finnhubKey`, and `newsApi` variables with values I'd acquired.

The **other** compilation errors came after the above fix.  They had to do with the **dio** package (`...dio_http_headers.dart:55:8: Error: The method 'DioHttpHeaders.add' has fewer named arguments than those of overridden method 'HttpHeaders.add'.`)  After I upgraded dio (See https://stackoverflow.com/questions/60476264/i-can-not-run-my-app-after-upgraded-flutter) the errors disappeared and everything worked great.
